manila/share/drivers/netapp/dataontap/cluster_mode/lib_multi_svm.py

```diff
@@ -611,17 +622,24 @@ def _delete_vserver_without_lock():
             self._client.delete_vserver(vserver,
                                         vserver_client,
                                         security_services=security_services)
-            ipspace_deleted = False
-            if (ipspace_name and ipspace_name not in CLUSTER_IPSPACES
-                    and not self._client.ipspace_has_data_vservers(
-                        ipspace_name)):
-                self._client.delete_ipspace(ipspace_name)
-                ipspace_deleted = True
-
-            if not ipspace_name or ipspace_deleted:
-                # NOTE(dviroel): only delete vlans if they are not being used
-                # by any ipspaces and data vservers.
-                self._delete_vserver_vlans(interfaces_on_vlans)
+
+            ipspace_deleted = self._client.delete_ipspace(ipspace_name)
+
+            # NOTE(dviroel): only delete vlans if they are not being used
+            # by any ipspaces and data vservers.
+            if ipspace_name and not ipspace_deleted:
```

Comment:
the if condition is opposite of itself before?
I think it's not necessary to check `ipspace_deleted` for deleting degraded port, as it's gone in neutron
what about the vlan belonging to current vserver?

Reply:
In case of `ipspace_deleted == True` we are deleting all vlans and don't care if they are degraded or belonging to the current vserver.
We don't want to miss any vlan, so that it would move the default ipspace and be left there

Reply:
I don't follow. You are indicating we delete vlans when ipspace can't be deleted?

Reply:
Yes, that is what we do during this if-condition here. We delete vlans that are degraded, i.e. orphan, and those that belong to the vserver we just deleted.

The method exits early if vserver delete raises, so we would never reach that point to accidentally delete a non-degraded vlan or a vlan of a non-deleted vserver.

So in conclusion we don't touch the vlans that are still in use - those belong to the data vservers that prevented the ipspace from being deleted.